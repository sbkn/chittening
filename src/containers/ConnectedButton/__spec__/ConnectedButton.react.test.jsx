import React from "react";
import renderer from "react-test-renderer";
import {ConnectedButton} from "../ConnectedButton.react.jsx";

describe("ConnectedButton", () => {

	it("should render correctly", () => {

		const tree = renderer.create(
			<ConnectedButton/>
		).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("should handle onClick", () => {

		const dispatchMock = jest.fn();

		const tree = renderer.create(
			<ConnectedButton dispatch={dispatchMock}/>
		).toJSON();

		tree.props.onClick();

		expect(dispatchMock.mock.calls.length).toBe(1);

	});
});
