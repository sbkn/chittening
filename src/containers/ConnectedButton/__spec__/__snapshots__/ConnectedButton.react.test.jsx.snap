// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedButton should render correctly 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Click me
</button>
`;
